### ISSUE TITLE
Ensure that OS-level font-size settings are honored in editor

### ISSUE THREAD

**Issue by iamthomasbishop** [wordpress-mobile/gutenberg-mobile#605]
While inspecting tablet layout refinements [here](https://github.com/wordpress-mobile/gutenberg-mobile/issues/604), I noticed that changes to font-size on the OS level aren't honored in the editor – both on iOS and Android.

## Steps to reproduce

1. Type a bunch of text in a Paragraph Block, preferably something in multiple lines.
2. Go to OS-level font-size settings, adjust to very large size.
3. Go back to Block Editor.
4. Notice that font-size is still 16.

**Comment by hotchemi** [comment]
@iamthomasbishop Right now I'm investigating the issue and let me confirm that this is the issue only for iOS or both iOS/Android🙇I built WP-iOS/Android app with the latest commit hash on develop branch and what I saw is the following.

### iOS

- [head commit](https://github.com/wordpress-mobile/WordPress-iOS/commit/958cccc5cebab22018c87af51f07db89fa5c57d6)

After changing system level font size I could reproduce the issue.

<img src="https://user-images.githubusercontent.com/471318/54942481-85baf900-4f72-11e9-8aa7-01bac379deae.gif" width="250" />

### Android

- [head commit](https://github.com/wordpress-mobile/WordPress-Android/commit/389ebd4c3e3e3f854767bfd929f1ea785723dd31)

<img src="https://user-images.githubusercontent.com/471318/54942555-adaa5c80-4f72-11e9-80ff-6a4c8683cf1c.gif" width="250" />

So right now I'm wondering this is the only issue for iOS..am I missing something?🙇Thank you!

**Comment by hypest** [comment]
👋 @hotchemi , judging from the gif you shared, I think that the editor there is Aztec, not Gutenberg. Have you perhaps tried Gutenberg too and the font sizing issue doesn't happen?

**Comment by hotchemi** [comment]
@hypest Thank you for pointing out🙇I tried again on Android and found gutenberg-mobile and wp-android behave slightly differently.

### gutenberg-mobile

On example app it seems works.

<img src="https://user-images.githubusercontent.com/471318/55007953-9ecfb280-5023-11e9-949f-263fff84c475.gif" width="250" />

### wp-android

I could repricate the issue.

<img src="https://user-images.githubusercontent.com/471318/55008098-db031300-5023-11e9-8c49-0d7bbf54286e.gif" width="250" />

**Comment by etoledom** [comment]
Hey @hotchemi - thank you for checking this out!

I have noticed issues on both platforms, including that red screen on android 🤔 
Also added this issue to the main Accessibility ticket to keep track of it:

https://github.com/wordpress-mobile/gutenberg-mobile/issues/476

**Comment by hotchemi** [comment]
@etoledom Thank you for clarification! After investigation turned out situation is a bit involved..😅

> including that red screen on android

This is another issue and after dismissing the red screen seems fontSize is changed properly on gutenberg-mobile example app.
But, the issue can be reproducible on wordpress-android and the cause is on native side I guess. https://github.com/wordpress-mobile/WordPress-Android/pull/9487 will resolve that part hopefully😃

**Comment by daniloercoli** [comment]
Android side PR: https://github.com/wordpress-mobile/WordPress-Android/pull/9509

**Comment by SergioEstevao** [comment]
So I tested this on WP-iOS and found the following:

- Some components of the editor adapt automatically when the setting is changed: The block picker, the media blocks placeholders. It looks that all components that are based on the standard RN (Text, Image)components work correctly
- Blocks that are based in Aztec RichText do not adapt automatically
- Paragraph, Heading, List, Code blocks adapt to the size if you exit the editor and open it again, or if you add them after the change of size is done
- Quote, PullQuote, Verse, Pre blocks never adapt
- The toolbars (block, editor) never adapt size.

Here are some screenshots:

<img src="https://user-images.githubusercontent.com/651601/84254534-42ae5c80-ab09-11ea-8dec-903862d4100d.png" width="300">
<img src="https://user-images.githubusercontent.com/651601/84254536-4346f300-ab09-11ea-8a06-1fed4460f3ad.png" width="300">

**Comment by etoledom** [comment]
Ouch, that doesn't look good 😬 

> Blocks that are based in Aztec RichText do not adapt automatically

This is probably because Attributed text? 🤔 

> Quote, PullQuote, Verse, Pre blocks never adapt

It's interesting that these blocks are also x-platform blocks. Do we have a default static font size for Aztec?

**Comment by SiobhyB** [comment]
Following https://github.com/WordPress/gutenberg/pull/57339, initial support for Dynamic Type will be introduced in 24.0 on iOS, due to be rolled out to users on January 22nd. I can verify that support for OS-level font-size changes is already there for Android.

I've also created a task list at https://github.com/WordPress/gutenberg/issues/57354 to follow up on enhancements, including enforcing a max/min size, refactoring RichText to automatically update, and reviewing other areas of the editor. I'll close this specific issue as we now have initial support and will track iterations elsewhere. 🙇‍♀️
